assets/arc-0031/server/routes/api/v1.ts

```diff
@@ -10,20 +9,38 @@ import users from '../../mock/users.db.json' assert { type: 'json' }
 interface AuthMessage {
   /** The domain name of the Verifier */
   domain: string
-  /** Algorand account to authenticate with */
+  /** Algorand account to authenticate with, encoded as a 32-bytes Algorand address */
   authAcc: string
-  /** Unique random nonce generated by the Verifier */
-  nonce: string
+  /** Unique random challenge of at least 15 characters and generated by the Verifier */
+  challenge: string
+  /** Algorand network identifier, encoded as a 32-bytes genesis hash of the network */
+  chainId: string
   /** Optional, description of the Verifier */
   desc?: string
   /** Optional, metadata */
   meta?: string
 }
 
+enum AlgorandGenesisHash {
+  MainNet = 'wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=',
+  TestNet = 'SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=',
+  BetaNet = 'mFgazF+2uRS1tMiL9dsj01hJGySEmPN28B/TjjvpVW0='
+}
+
```

Comment:
What's the advantage of using the `AlgorandGenesisHash` and not simply a string like 'MainNet', 'TestNet' or 'BetaNet' or a an enum like the one used by Pera Wallet? I feel like we should pick a consistent decision across the whole stack here.

```
enum AlgorandChainId {
  MainNet = 416001,
  TestNet = 416002,
  BetaNet = 416003
}
```

Reply:
These are different concepts. 

The enum `AlgorandChainId` is defined by PeraConnect to identify the chain and it has nothing to do with the identifiers used by the Algorand protocol. We cannot enforce those values because depend on PeraConnect implementation and might change arbitrarily.

The enum `AlgorandGenesisHash` is the **unique** genesis hash that identifies an Algorand network (see [here](https://developer.algorand.org/docs/get-details/algorand-networks/mainnet/#genesis-hash)). It is defined at the protocol level and will never change (it is the hash of the genesis block of the network).

The former is a PeraConnect requirement, the latter is an ARC31 protocol requirement.